sonyapilib/device.py

```diff
@@ -359,7 +358,10 @@ def register(self):
 
         else:
             raise ValueError(
-                "Regisration mode {0} is not supported".format(registration_action.mode))
+                "Registration mode {0} is not supported".format(registration_action.mode))
+
+        if registration_result == AuthenicationResult.SUCCESS:
```

Comment:
This will only be reached for generation 2 devices as in every other device it will always return PIN_NEEDED the implementation for api version 4 is faulty. 
This has been changed in my current branch to implement version 4 of the API. 
You can make this call in send authentication.

Reply:
I added the function call in `send_authentication()`, but that doesn't get called for Gen2 devices ;-)